### PR TITLE
Fix embedded images and list numbering, cleanup image captions in the docs

### PR DESCRIPTION
**Pull request by nightlark**
### Description

If merged this pull request will fix embedded images (PDF->PNG), fix formatting to get correct list numbers on several pages, and change some of the image captions/alt-text so they look less like badly formed HTML mistakes.

### Proposed changes
- Change some image captions to better describe the image so that they look less like bad html
- Fixes some other misc markdown issues with headers
- Fix the numbered lists by changing indentation on pages such as https://helics.readthedocs.io/en/latest/user-guide/helics_co-sim_sequence.html and https://helics.readthedocs.io/en/latest/developer-guide/style.html
- Fix embedded images on user guide pages (PDFs show up as broken image links) such as https://helics.readthedocs.io/en/latest/user-guide/helics_key_concepts.html, https://helics.readthedocs.io/en/latest/user-guide/value_federates.html, https://helics.readthedocs.io/en/latest/user-guide/message_federates.html, https://helics.readthedocs.io/en/latest/user-guide/filters.html, https://helics.readthedocs.io/en/latest/user-guide/timing.html

### CODE REVIEW

File: docs/user-guide/message_federates.md [hunk 1]
```diff
@@ -10,13 +10,13 @@ As previously discussed, message federates interact with the federation by defin
...
 
 Consider a counter-example: automated meter-reading (AMI) using a wireless network that connects all meters in a distribution system to a data-aggregator in the substation (where, presumably, the data travels over a dedicated wired connection to a control room). All meters will have a single endpoint over which they will send their data but what about the receiver? The co-simulation could be designed with the data-aggregator having a unique endpoint for each meter but this implies come kind of dedicated communication channel for each meter; this is not the case with wireless communication. Instead, it is probably better to create a single endpoint representing the wireless connection the data aggregator has with the AMI network. In this case, messages from any of the meters in the system will be flowing through the same endpoint and to differentiate the messages from each other, the federate will have to be written to examine the metadata with the message to determine its original source.
 
-![ami message federates](../img/ami_message_federate.pdf)
+![ami message federates](../img/ami_message_federate.png)
```
Comment (phlptp) [inline docs/user-guide/message_federates.md]: I don't have any issue with it, but what was the reason for changing to png?

File: docs/user-guide/simulator_integration.md [hunk 2]
```diff
@@ -2,20 +2,20 @@
...
-   - Octave
-   - C# (somewhat limited as of yet)
+      The languages currently supported by HELICS are:
+      - C++
+      - C
+      - Python (2 and 3)
+      - Java
+      - MATLAB
+      - Octave
+      - C# (somewhat limited as of yet)
```
Comment (phlptp) [inline docs/user-guide/simulator_integration.md]: Add some of the language Dheepak was working with Julia or nim(@kdheepak) any more?

File: docs/user-guide/message_federates.md (hunk 1 above)
Comment (nightlark) [reply docs/user-guide/message_federates.md]: The PDFs just show up as the unable to load image icon (little box with an X in it) and don't show up (at least in the recent version of Chrome I'm using).

Comment (nightlark) [reply docs/user-guide/message_federates.md]: As I was going through RTD, it felt like some of the material tries to be for everyone — wondering if separating some of it out into a “co-sim primer” and “helics by example” type resources would help make it feel less scattered.

File: docs/user-guide/simulator_integration.md (hunk 2 above)
Comment (nightlark) [reply docs/user-guide/simulator_integration.md]: Added Julia and Nim to the list.

**Comment by phlptp** [comment]
you may want to cancel all but the last of fix-rtd builds on appveyor.  It could take a while to work through them.

**Comment by nightlark** [comment]
Good idea. Got it right before last of the message branch builds finished.

**Comment by nightlark** [comment]
When I get back in a few hours I’m going to look over the pages one last time for more formatting issues.

**Comment by phlptp** [comment]
@nightlark if you make any more edits,  or not can you trigger another round of on the message timer tests.    I hate tracking down these sporadic failures that can never be replicated on local machines.

**Comment by nightlark** [comment]
I'm calling this done now -- looking over pages "one last time" kept uncovering formatting issues or broken links.

**Pull request merged**